### PR TITLE
add new subcommand for proxy-config to check the ROOTCA between 2 given pods

### PR DESCRIPTION
**Pull request by zhlsunshine**
This PR will add new subcommand for proxy-config of istioctl to check their ROOTCA between 2 given pods.

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: istioctl/cmd/proxyconfig.go [hunk 1]
```diff
@@ -946,6 +946,77 @@ func secretConfigCmd() *cobra.Command {
 	return secretConfigCmd
 }
 
+func rootCACompareConfigCmd() *cobra.Command {
+	var podName1, podName2, podNamespace1, podNamespace2 string
+
+	rootCACompareConfigCmd := &cobra.Command{
+		Use:   "rootca-comp [pod/]<name-1>[.<namespace-1>] [pod/]<name-2>[.<namespace-2>]",
+		Short: "Compare ROOTCA values for given 2 pods",
```
Comment (irisdingbj) [inline istioctl/cmd/proxyconfig.go:954]: Compare ROOTCA values for the two given pods

File: istioctl/cmd/proxyconfig.go [hunk 2]
```diff
@@ -946,6 +946,77 @@ func secretConfigCmd() *cobra.Command {
...
+		Long:  `Compare ROOTCA values for given 2 pods to check the connectivity availability between them.`,
```
Comment (irisdingbj) [inline istioctl/cmd/proxyconfig.go:955]: Compare ROOTCA values for given 2 pods to check the connectivity  between them

File: istioctl/cmd/proxyconfig.go [hunk 3]
```diff
@@ -946,6 +946,77 @@ func secretConfigCmd() *cobra.Command {
...
+		Example: `  # Compare ROOTCA values for given 2 pods to check the connectivity availability between them.
+  istioctl proxy-config rootca-comp <pod-name-1[.namespace]> <pod-name-2[.namespace]>`,
+		Aliases: []string{"rootca-comp", "rc"},
+		Args: func(cmd *cobra.Command, args []string) error {
+			if (len(args) <= 2) != (configDumpFile == "") {
+				cmd.Println(cmd.UsageString())
+				return fmt.Errorf("see the usage above, rootca-comp requires 2 pods as arguments")
```
Comment (shamsher31) [inline istioctl/cmd/proxyconfig.go:962]:
```suggestion
				return fmt.Errorf("rootca-comp requires 2 pods as an argument")
```

File: istioctl/cmd/proxyconfig.go [hunk 4]
```diff
@@ -946,6 +946,77 @@ func secretConfigCmd() *cobra.Command {
...
+				if podName2, podNamespace2, err = getPodName(args[1]); err != nil {
+					return err
+				}
+				configWriter2, err = setupPodConfigdumpWriter(podName2, podNamespace2, c.OutOrStdout())
+				if err != nil {
+					return err
+				}
+			} else {
+				c.Println(c.UsageString())
+				return fmt.Errorf("please see the usage above and make sure that there are 2 pods for the comparison")
```
Comment (shamsher31) [inline istioctl/cmd/proxyconfig.go:987]:
```suggestion
				return fmt.Errorf("rootca-comp requires 2 pods as an argument")
```

File: istioctl/cmd/proxyconfig.go (hunk 3 above)
Comment (zhlsunshine) [reply istioctl/cmd/proxyconfig.go:962]: thanks @shamsher31, done!

File: istioctl/cmd/proxyconfig.go (hunk 4 above)
Comment (zhlsunshine) [reply istioctl/cmd/proxyconfig.go:987]: thanks @shamsher31, done!

File: istioctl/cmd/proxyconfig.go (hunk 1 above)
Comment (zhlsunshine) [reply istioctl/cmd/proxyconfig.go:954]: thanks @irisdingbj done!

File: istioctl/cmd/proxyconfig.go (hunk 2 above)
Comment (zhlsunshine) [reply istioctl/cmd/proxyconfig.go:955]: thanks @irisdingbj done!

File: istioctl/cmd/proxyconfig.go [hunk 5]
```diff
@@ -946,6 +946,77 @@ func secretConfigCmd() *cobra.Command {
 	return secretConfigCmd
 }
 
+func rootCACompareConfigCmd() *cobra.Command {
+	var podName1, podName2, podNamespace1, podNamespace2 string
+
+	rootCACompareConfigCmd := &cobra.Command{
+		Use:   "rootca-comp [pod/]<name-1>[.<namespace-1>] [pod/]<name-2>[.<namespace-2>]",
```
Comment (shamsher31) [inline istioctl/cmd/proxyconfig.go:953]: I think the command should be `rootca-cmp`? Just an idea from https://github.com/google/go-cmp

```suggestion
		Use:   "rootca-cmp [pod/]<name-1>[.<namespace-1>] [pod/]<name-2>[.<namespace-2>]",
```

Comment (zhlsunshine) [reply istioctl/cmd/proxyconfig.go:953]: sure, that's a good idea!

**Comment by zhlsunshine** [comment]
Here is the output for verification below:

```
$ istioctl proxy-config rootca-compare httpbin-5f76c56644-5x7tt.foo sleep-7bbcb459d8-57j2w.foo
Both [httpbin-5f76c56644-5x7tt.foo] and [sleep-7bbcb459d8-57j2w.foo] have the identical ROOTCA, theoretically the connectivity between them is available
```

```
$ istioctl proxy-config rootca-compare httpbin-5f76c56644-5x7tt.foo httpbin-5f76c56644-qfwkm.bar
Error: Both [httpbin-5f76c56644-5x7tt.foo] and [httpbin-5f76c56644-qfwkm.bar] have the non identical ROOTCA, theoretically the connectivity between them is unavailable
```

**Review by irisdingbj — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**